homeassistant/components/econet/__init__.py

```diff
@@ -1 +1,171 @@
-"""The econet component."""
+"""Support for EcoNet products."""
+import asyncio
+from datetime import timedelta
+import logging
+
+from pyeconet import EcoNetApiInterface
+from pyeconet.equipment import EquipmentType
+from pyeconet.errors import InvalidCredentialsError, PyeconetError
+
+from homeassistant.config_entries import SOURCE_IMPORT
+from homeassistant.const import CONF_EMAIL, CONF_PASSWORD
+from homeassistant.core import callback
+from homeassistant.exceptions import ConfigEntryNotReady
+from homeassistant.helpers.dispatcher import dispatcher_send
+from homeassistant.helpers.entity import Entity
+from homeassistant.helpers.event import async_track_time_interval
+
+from .const import API_CLIENT, DOMAIN, EQUIPMENT
+
+_LOGGER = logging.getLogger(__name__)
+
+PLATFORMS = ["water_heater", "binary_sensor", "sensor"]
+PUSH_UPDATE = "econet.push_update"
+
+INTERVAL = timedelta(minutes=60)
+
+
+async def async_setup(hass, config):
+    """Set up the EcoNet component."""
+
+    hass.data[DOMAIN] = {}
+    hass.data[DOMAIN][API_CLIENT] = {}
+    hass.data[DOMAIN][EQUIPMENT] = {}
+
+    if DOMAIN not in config:
+        return True
+
+    conf = config[DOMAIN]
+
+    hass.async_create_task(
```

Comment:
There was no config yaml under the econet key just under the water heater platform. So we should import the config from that platform instead.

Reply:
The reason I moved it to econet was because their API supports thermostats as well. 

I know you said to add it and make it deprecated, but it seems kinda pointless. The current integration is 100% broken, and anyone I have found with issue is in a discord group using this as a custom component currently so it seems like a waste I think? Thoughts?

Reply:
Do you mean that the old credentials don't work with the new API? I'm not following.

Reply:
Well actually I am not sure. I didn't own a device when the old API was being used, so I have only ever been able to use the new API with my personal setup. The old API doesn't work at all now.

So it is possible that the old credentials that some users have will work with the new API but I'm not 100% and I would be surprised if anyone still has the old setup in their config since it would just be filling their logs with errors at this point.

Reply:
If we don't think it makes sense we can skip the import to config flow and ask users to configure the integration from scratch, in the breaking change paragraph.

Reply:
Yeah, I think that makes the most sense here.

Reply:
Updated the breaking changes paragraph, I also created a new MR for .io because the originally linked one wasn't accurate.

Reply:
Then we should remove the import to config flow here.

Reply:
Oh I see, I didn't realize config yaml wasn't allowed. 

Where can I find those ADRs?

Reply:
In our architecture repo.

Reply:
Awesome, good to know just read through it. 

I'll get this removed.